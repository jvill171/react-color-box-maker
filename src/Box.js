import React from "react";

const Box = ({id, backgroundColor, width=5, height=5, handleRemove}) => {

    return (
        <div
            className="box"
            id={id}
            style={{
                backgroundColor,
                width:`${width}em`,
                height:`${height}em`,
                border:"1px solid black", margin:"5px"
            }}
        >
            <button onClick={handleRemove}>X</button>
        </div>
    )
}

export default Box;