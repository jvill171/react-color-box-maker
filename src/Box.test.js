import { render } from "@testing-library/react"
import Box from './Box';

// Smoke test
test('renders without crashing', () => {
  render(<Box />);
});

  
// Snapshot test
it("matches snapshot", ()=>{
    const { asFragment } = render(
        <Box/>
    );
    expect(asFragment()).toMatchSnapshot()
})