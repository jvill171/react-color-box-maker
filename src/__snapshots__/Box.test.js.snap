// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="box"
    style="width: 5em; height: 5em; border: 1px solid black; margin: 5px;"
  >
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
